## Identifying candidate genetic variants for egg number by analyzing over 1000 fully sequenced layers

Repo for code used in paper

### Overview

This document provides step-by-step guidance on key analyses, including variant calling, GWAS, eQTL mapping, transcriptome-wide association studies, and Mendelian randomization. 

1. Variant Calling from Whole Genome Sequencing (WGS)
This file covers:

Data Preparation: Raw sequence data processing and quality control.
Alignment: Aligning reads to the reference genome.
Variant Calling: Identification of SNPs and indels using appropriate tools.
Post-Processing: Variant filtering and annotation.
Output: Overview of final output files and formats.
2. Genome-Wide Association Study (GWAS)
This file details the steps for GWAS using both additive and additive-dominance models, including:

Data Preparation: Phenotype and genotype data preprocessing.
Statistical Models: Applying the additive and additive-dominance models.
Results Interpretation: Handling significant results, with multiple-testing correction.
Output: Descriptions of result files and interpretation guidance.
3. eQTL Mapping
This file includes instructions for eQTL mapping, covering:

Data Preparation: Preparing expression and genotype data.
Statistical Testing: Model fitting and testing for eQTLs.
Significant eQTLs: Post-analysis filtering of significant eQTLs.
Visualization: Optional steps for visualizing eQTL relationships.
Output: Key outputs and their interpretation.
4. Transcriptome-Wide Association Study (TWAS)
This file outlines TWAS procedures:

Preliminary Steps: Preparation of reference transcriptome data.
TWAS Analysis: Model training, imputation, and association testing.
Post-Processing: Significant associations and functional insights.
Output: Format and meaning of TWAS results.
5. Mendelian Randomization (MR) Analysis
This file provides an overview of the MR analysis:

Instrument Selection: Choosing genetic instruments for MR.
Data Harmonization: Ensuring alignment between exposure and outcome datasets.
MR Analysis: Statistical models used for causal inference.
Sensitivity Analysis: Testing robustness of MR findings.
Output: Summary of MR results and their interpretation.

Markdown file outlining steps taken to call variants from WGS data is here 

Markdown file outlining steps taken to do genome-wide assocation study with additive and additive-dominance model is here 

Markdown file outlining steps taken to do eQTL-mapping is here  

Markdown file outlining steps taken to do transcrioptome-wide association study is here  

Markdown file outlining steps taken to do Mendelian randomization analysis is here  

#### Contact
Contact information for author questions or feedback.

###### Aixin Ni

naixin951@163.com
[https://orcid.org/0000-0002-4478-915X](https://orcid.org/my-orcid?orcid=0009-0002-7938-1231)
