## Identifying candidate genetic variants for egg number by analyzing over 1000 fully sequenced layers

Repo for code used in paper

### Overview

This document provides step-by-step guidance on key analyses, including variant calling, GWAS, eQTL mapping, transcriptome-wide association studies, and Mendelian randomization. 

1. Variant Calling from Whole Genome Sequencing (WGS)
Markdown file outlining steps taken to call variants from WGS data is [here 1_genotype_calling](1_genotype_calling/genotype_calling.md)

2. Genome-Wide Association Study (GWAS)
Markdown file outlining steps taken to do genome-wide assocation study with additive and additive-dominance model is [here 2_GWAS](2_GWAS/2_GWAS.md) 

3. eQTL Mapping
Markdown file outlining steps taken to do eQTL-mapping is [here 3_eQTL_mapping](3_eQTL_mapping/3_eQTL_mapping.md) 

4. Transcriptome-Wide Association Study (TWAS)
Markdown file outlining steps taken to do transcrioptome-wide association study is [here 4_TWAS](4_TWAS/4_TWAS.md) 

5. Mendelian Randomization (MR) Analysis
Markdown file outlining steps taken to do Mendelian randomization analysis is [here 5_MR](5_MR/5_MR.md) 

#### Contact
Contact information for author questions or feedback.

###### Aixin Ni

naixin951@163.com

[https://orcid.org/0000-0002-4478-915X](https://orcid.org/my-orcid?orcid=0009-0002-7938-1231)
